cdap-api/src/main/java/co/cask/cdap/api/artifact/PluginConfigurer.java

```diff
@@ -25,7 +26,7 @@
  * This interface provides methods to register plugin usage in a CDAP Program. The registered plugins will be
  * available at the runtime of the Program.
  */
-public interface PluginConfigurer {
+public interface PluginConfigurer extends DatasetConfigurer {
```

Comment:
I think it makes sense if the plugin is available across the entire app instead of local to a program. It is more consistent with the stream and dataset behavior.  Also, we have a use case in data quality where an aggregation function plugin is used in both a mapreduce program and a service program. In that case it's easier to register it once in the app versus in both programs.